src/Compilation.zig

```diff
@@ -855,14 +855,31 @@ pub fn create(gpa: Allocator, options: InitOptions) !*Compilation {
             break :blk false;
         };
 
-        const sysroot = blk: {
-            if (options.sysroot) |sysroot| {
-                break :blk sysroot;
-            } else if (options.native_darwin_sdk) |sdk| {
-                break :blk sdk.path;
-            } else {
-                break :blk null;
+        const darwin_native = (comptime builtin.target.isDarwin()) and options.target.isDarwin() and
+            options.sysroot == null;
+        const darwin_sdk: ?std.zig.system.darwin.DarwinSDK = if (darwin_native and
+            !std.zig.system.NativePaths.isNix(arena) and
+            std.zig.system.darwin.isDarwinSDKInstalled(arena))
+            std.zig.system.darwin.getDarwinSDK(arena, options.target)
+        else
+            null;
+        const want_native_paths = (options.sysroot == null and options.is_native_os and
+            (options.system_lib_names.len > 0 or options.want_native_include_dirs orelse false)) or
+            darwin_native;
+        const native_paths = if (want_native_paths) blk: {
+            const paths = std.zig.system.NativePaths.detect(arena, options.target) catch |err| {
+                fatal("unable to detect native system paths: {s}", .{@errorName(err)});
+            };
+            for (paths.warnings.items) |warning| {
+                log.warn("{s}", .{warning});
```

Comment:
can you give an example of what this looks like when it occurs?

Reply:
I cannot, I copied the code that was already there. You may either ask the original author, or we can remove it.